### ISSUE TITLE
MetaRefresh deletes metadata when the src is not reachable

### ISSUE THREAD

**Issue by ymakhloufi** [simplesamlphp/simplesamlphp-module-metarefresh#7]
UseCase:
1. The MetaRefresh module successfully refreshed the metadata at 00:00 midnight and writes the metadata to disk.
2. The MetaRefresh module attempts to refresh the metadata at 06:00 AM but the server is unavailable (e.g. DNS issue), so the refresh fails
3. As a result all previously stored metadata is lost (which is REALLY bad, because all SAML requests will now fail!)

As a workaround, we have currently modified the cron entry to first ensure that the src is available (curl) and only if that is the case, refresh the metadata. But that's a nasty workaround and also means we have to maintain the URL in two different places.

**Comment by thijskh** [comment]
This is a bug. The code does intend to do what you do. So we need to find out why it doesn't.

**Comment by tvdijen** [comment]
I don't think it's a bug.. The code will try to save fresh metadata and then as a last step any expired metadata is removed,..

**Comment by thijskh** [comment]
Metadata usually does not expire within 6 hours I think.

**Comment by thijskh** [comment]
I have tried to reproduce this, also specifically with name resolving, but it works fine for me (now). It can be that we fixed the problem in the mean time. There was at least an issue with refreshing it through the web UI that did not properly handle this, but I addressed that last week by deduplicating the code (#19).

Indeed it's possible that metadata disappears when expired but as I said metadata should have a long shelf life. In any case I've submitted #20 to make this expiry problem more obvious when you hit it.

For now I consider this to be not/no longer a problem.
If the problem still persists with latest master we'd need some more information about how to reproduce it exactly (which config).
